modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -751,3 +753,26 @@
                   (not host) (assoc :host (str account ".snowflakecomputing.com"))
                   (not port) (assoc :port 443))]
     (driver/incorporate-ssh-tunnel-details :sql-jdbc details)))
+
+(defmethod sql-jdbc.sync/describe-fields-sql :snowflake
+  [driver & {:keys [schema-names table-names details]}]
+  (sql/format {:select [[:c.COLUMN_NAME :name]
+                        [[:- :c.ORDINAL_POSITION [:inline 1]] :database-position]
+                        [:c.TABLE_SCHEMA :table-schema]
+                        [:c.TABLE_NAME :table-name]
+                        [[:upper :c.DATA_TYPE] :database-type]
+                        [[:!= :c.IDENTITY_GENERATION nil] :database-is-auto-increment]
+                        [[:and
+                          [:or [:= :COLUMN_DEFAULT nil] [:= [:lower :COLUMN_DEFAULT] [:inline "null"]]]
+                          [:= :IS_NULLABLE [:inline "NO"]]
+                          [:not [:!= :c.IDENTITY_GENERATION nil]]]
+                         :database-required]
+                        [[:= :c.IS_IDENTITY [:inline "YES"]] :pk?]
+                        [:c.*]
+                        [[:nullif :c.COMMENT [:inline ""]] :field-comment]]
+               :from [[[:raw (format "\"%s\".\"%s\".\"%s\"" (:db details) "INFORMATION_SCHEMA" "COLUMNS")] :c]]
```

Comment:
I think you should be able to use `h2x/identifier` here to generate this sort of quoted identifier